From 5ad41eef0be517715a940b788d9c20ca4fb53905 Mon Sep 17 00:00:00 2001
From: Nobuhiro Iwamatsu <iwamatsu@nigauri.org>
Date: Sat, 4 Jun 2011 17:38:58 -0700
Subject: Update lengthen time-out time for t/re/re.t.

Bug-Debian: http://bugs.debian.org/626125
Bug: http://rt.perl.org/rt3/Public/Bug/Display.html?id=92226
Origin: upstream, http://perl5.git.perl.org/perl.git/commitdiff/7c47ab277d9951f7d3b1f59075afbc754894081c

When we carry out this test on SH4, it becomes the time-out.
2 seconds are set in watchdog, but are too short for SH4.
This patch was changed for 10 seconds.

$ time ./perl  t/re/re.t
1..19
ok 1 - is_regexp(REGEXP ref)
ok 2 - is_regexp(REGEXP)
ok 3 - is_regexp("")
ok 4 - regexp_pattern[0] (ref)
ok 5 - regexp_pattern[1] (ref)
ok 6 - scalar regexp_pattern (ref)
ok 7 - regexp_pattern[0] (bare REGEXP)
ok 8 - regexp_pattern[1] (bare REGEXP)
ok 9 - scalar regexp_pattern (bare REGEXP)
ok 10 - !regexp_pattern("")
ok 11 - regnames
ok 12 - regnames
ok 13 - regnames in scalar context
ok 14 - regnames
ok 15
ok 16
ok 17
ok 18
ok 19 - Didn't loop

real    0m7.482s
user    0m3.848s
sys     0m0.036s

Patch-Name: fixes/re.t_sh4_timeout.diff
---
 ext/re/t/re_funcs_u.t |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/ext/re/t/re_funcs_u.t b/ext/re/t/re_funcs_u.t
index dcb35e1..57d7281 100644
--- a/ext/re/t/re_funcs_u.t
+++ b/ext/re/t/re_funcs_u.t
@@ -127,7 +127,7 @@ if ('1234'=~/(?:(?<A>\d)|(?<C>!))(?<B>\d)(?<A>\d)(?<B>\d)/){
 
     { # Keep these tests last, as whole script will be interrupted if times out
         # Bug #72998; this can loop 
-        watchdog(2);
+        watchdog(10);
         eval '"\x{100}\x{FB00}" =~ /\x{100}\N{U+66}+/i';
         pass("Didn't loop");
 
